test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -566,9 +566,113 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			By("Saving the old performance profile")
 			initialProfile = profile.DeepCopy()
 		})
+		When("Realtime enabled", func() {
+			It("[test_id:50990][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Should updated kernel arguments and tuned", func() {
+				By("Modifying profile")
+				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
+					HighPowerConsumption: pointer.Bool(false),
+					RealTime:             pointer.Bool(true),
+				}
+				By("Updating the performance profile")
+				profiles.UpdateWithRetry(profile)
```

Comment:
I have caught up the error. The issue is because PAO doesn't detect the change in WorkloadHints, the new performanceProfile is applied but mcp does not care about it. One fast fix is also change cpus like that:
```
				By("Modifying profile")
				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
					HighPowerConsumption: pointer.Bool(false),
					RealTime:             pointer.Bool(true),
				}

				reserved := performancev2.CPUSet("0,1")
				isolated := performancev2.CPUSet("1,2")
				profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
				Expect(err).ToNot(HaveOccurred())

				profile.Spec.CPU = &performancev2.CPU{
					Reserved: &reserved,
					Isolated: &isolate
```
but also I have to investigate why mcp does not change to updating when performanceProfile changes.